File: packages/sdks/python/tester-local/tests/test_create_patient.py
import os

from generated.client import Metriport
from generated import UsState, Address
from generated import commons
from generated.medical import BasePatient, PersonalIdentifier_DriversLicense, PersonalIdentifier_Ssn

import os
from dotenv import load_dotenv

load_dotenv()

api_key = os.environ.get("API_KEY")
facility_id = os.environ.get("FACILITY_ID")
base_url = os.environ.get("BASE_URL")


def test_create_patient() -> None:
    metriport = Metriport(api_key=api_key, base_url=base_url)
    response = metriport.medical.patient.create(
      facility_id=facility_id,
      first_name="John",
        last_name="Doe",
        dob="1980-01-01",
        gender_at_birth="M",
        personal_identifiers=[
            PersonalIdentifier_DriversLicense(
                type="driversLicense",
                state='CA',
                value="12345678",
            ),
            PersonalIdentifier_Ssn(
                type="ssn",
                value="123456789",
            )
        ],
        address=[Address(
            address_line_1="123 Main St",
            city="Los Angeles",
            state='CA',
            zip="90001",
            country="USA"
        )]
    )
    print(f"Received patient with ID: {response.id}")

File: packages/sdks/java/tester-local/src/test/java/com.metriport/TestCreatePatient.java
package com.metriport;

import com.metriport.generated.Metriport;
import com.metriport.generated.resources.medical.patient.requests.PatientCreate;
import com.metriport.generated.resources.medical.patient.types.BasePatient;
import com.metriport.generated.resources.medical.patient.types.PersonalIdentifier;
import com.metriport.generated.resources.medical.patient.types.DriversLicense;
import com.metriport.generated.resources.medical.patient.types.Ssn;
import com.metriport.generated.resources.commons.types.Address;
import com.metriport.generated.resources.commons.types.UsState;
import io.github.cdimascio.dotenv.Dotenv;
import org.junit.jupiter.api.Test;

import java.util.Collections;
import java.util.Arrays;

public class TestCreatePatient {
    @Test
    public void createPatient() {
        Dotenv dotenv = Dotenv.load();

        String apiKey = dotenv.get("API_KEY");
        String facilityId = dotenv.get("FACILITY_ID");
        String baseUrl = dotenv.get("BASE_URL");

        Metriport client = Metriport.builder()
                .apiKey(apiKey)
                .url(baseUrl)
                .build();

        DriversLicense driversLicense = DriversLicense.builder()
                .state(UsState.CA)
                .value("12345678")
                .build();

        PersonalIdentifier personalIdentifierDL = PersonalIdentifier.driversLicense(driversLicense);

        Ssn ssn = Ssn.builder()
                .value("123456789")
                .build();

        PersonalIdentifier personalIdentifierSSN = PersonalIdentifier.ssn(ssn);

        BasePatient patientData = BasePatient.builder()
                .firstName("John")
                .lastName("Doe")
                .dob("1980-01-01")
                .genderAtBirth("M")
                .personalIdentifiers(Arrays.asList(personalIdentifierDL, personalIdentifierSSN))
                .address(Collections.singletonList(
                        Address.builder()
                                .addressLine1("123 Main St")
                                .city("Los Angeles")
                                .state(UsState.CA)
                                .zip("90001")
                                .country("USA")
                                .build()
                ))
                .build();

        PatientCreate request = PatientCreate.builder()
                .facilityId(facilityId)
                .body(patientData)
                .build();

        var response = client.medical().patient().create(request);
        System.out.println("Received patient with ID: " + response.getId());
    }
}
